perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/MuadmPasswordManagerModule.java

```diff
@@ -8,10 +8,17 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.regex.Pattern;
+
 public class MuadmPasswordManagerModule extends GenericPasswordManagerModule {
 
 	private final static Logger log = LoggerFactory.getLogger(MuadmPasswordManagerModule.class);
 
+	protected final Pattern muadmPasswordContainsDigit = Pattern.compile(".*[0-9].*");
+	protected final Pattern muadmPasswordContainsLower = Pattern.compile(".*[a-z].*");
+	protected final Pattern muadmPasswordContainsUpper = Pattern.compile(".*[A-Z].*");
+	protected final Pattern muadmPasswordContainsSpec = Pattern.compile(".*[\\x20-\\x2F\\x3A-\\x40\\x5B-\\x60\\x7B-\\x7E].*");
```

Comment:
If you use these values only in this class, you can make them private. Also, you can make them static.

Also, the names can be simplified. You don't need to name them with muadmPassword, since this is already present in the class name.
```suggestion
	private static final Pattern digitPattern = Pattern.compile(".*[0-9].*");
	private static final Pattern lowerPattern = Pattern.compile(".*[a-z].*");
	private static final Pattern upperPattern = Pattern.compile(".*[A-Z].*");
	private static final Pattern specCharsPattern = Pattern.compile(".*[\\x20-\\x2F\\x3A-\\x40\\x5B-\\x60\\x7B-\\x7E].*");
```